App/ViewModels/AgentViewModel.cs

```diff
@@ -84,15 +107,56 @@ public partial class AgentViewModel : ObservableObject, IModelUpdateable<AgentVi
 
     public readonly Uuid Id;
 
+    // This is set only for "dummy" agents that represent unstarted workspaces. If set, then ConfiguredFqdn
+    // should be empty, otherwise it will override this.
+    [ObservableProperty]
+    [NotifyPropertyChangedFor(nameof(ViewableHostname))]
+    [NotifyPropertyChangedFor(nameof(ViewableHostnameSuffix))]
+    [NotifyPropertyChangedFor(nameof(FullyQualifiedDomainName))]
+    public required partial string ConfiguredHostname { get; set; }
+
+    // This should be set if we have an FQDN from the VPN service, and overrides ConfiguredHostname if set.
     [ObservableProperty]
-    [NotifyPropertyChangedFor(nameof(FullHostname))]
-    public required partial string Hostname { get; set; }
+    [NotifyPropertyChangedFor(nameof(ViewableHostname))]
+    [NotifyPropertyChangedFor(nameof(ViewableHostnameSuffix))]
+    [NotifyPropertyChangedFor(nameof(FullyQualifiedDomainName))]
+    public required partial string ConfiguredFqdn { get; set; }
 
     [ObservableProperty]
-    [NotifyPropertyChangedFor(nameof(FullHostname))]
-    public required partial string HostnameSuffix { get; set; } // including leading dot
+    [NotifyPropertyChangedFor(nameof(ViewableHostname))]
+    [NotifyPropertyChangedFor(nameof(ViewableHostnameSuffix))]
+    [NotifyPropertyChangedFor(nameof(FullyQualifiedDomainName))]
+    public required partial string ConfiguredHostnameSuffix { get; set; }
+
+
+    public string FullyQualifiedDomainName
+    {
+        get
+        {
+            if (!string.IsNullOrEmpty(ConfiguredFqdn)) return ConfiguredFqdn;
+            return ConfiguredHostname + ConfiguredHostnameSuffix;
+        }
+    }
 
-    public string FullHostname => Hostname + HostnameSuffix;
+    /// <summary>
+    /// ViewableHostname is the hostname portion of the fully qualified domain name (FQDN) specifically for
+    /// views that render it differently than the suffix. If the ConfiguredHostnameSuffix doesn't actually
+    /// match the FQDN, then this will be the entire FQDN, and ViewableHostnameSuffix will be empty.
+    /// </summary>
+    public string ViewableHostname => !FullyQualifiedDomainName.EndsWith(ConfiguredHostnameSuffix)
+        ? FullyQualifiedDomainName
+        : FullyQualifiedDomainName.Substring(0,
+            FullyQualifiedDomainName.Length - ConfiguredHostnameSuffix.Length);
```

Comment:
Can you do `FullyQualifiedDomainName[0..^ConfiguredHostnameSuffix.Length]` instead?

Reply:
That's nicer. AIs and/or the Stack Overflow articles they search don't know about that cuz it's a new(ish) feature of C#.